fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RemoveAggregationFromAggTable.java

```diff
@@ -0,0 +1,115 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+package com.starrocks.sql.optimizer.rule.transformation;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.starrocks.catalog.Column;
+import com.starrocks.catalog.FunctionSet;
+import com.starrocks.catalog.MaterializedIndexMeta;
+import com.starrocks.catalog.OlapTable;
+import com.starrocks.catalog.PartitionInfo;
+import com.starrocks.catalog.RangePartitionInfo;
+import com.starrocks.sql.optimizer.OptExpression;
+import com.starrocks.sql.optimizer.OptimizerContext;
+import com.starrocks.sql.optimizer.operator.OperatorType;
+import com.starrocks.sql.optimizer.operator.logical.LogicalAggregationOperator;
+import com.starrocks.sql.optimizer.operator.logical.LogicalOlapScanOperator;
+import com.starrocks.sql.optimizer.operator.logical.LogicalProjectOperator;
+import com.starrocks.sql.optimizer.operator.pattern.Pattern;
+import com.starrocks.sql.optimizer.operator.scalar.CallOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
+import com.starrocks.sql.optimizer.rewrite.ReplaceColumnRefRewriter;
+import com.starrocks.sql.optimizer.rule.RuleType;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class RemoveAggregationFromAggTable extends TransformationRule {
+    private static List<String> unsupportedFunctionNames =
+            ImmutableList.of(FunctionSet.BITMAP_UNION,
+                    FunctionSet.BITMAP_UNION_COUNT,
+                    FunctionSet.HLL_UNION,
+                    FunctionSet.HLL_UNION_AGG,
+                    FunctionSet.PERCENTILE_UNION);
+
+    public RemoveAggregationFromAggTable() {
+        super(RuleType.TF_REMOVE_AGGREGATION_BY_AGG_TABLE,
+                Pattern.create(OperatorType.LOGICAL_AGGR).addChildren(Pattern.create(
+                        OperatorType.LOGICAL_OLAP_SCAN)));
+    }
+
+    @Override
+    public boolean check(OptExpression input, OptimizerContext context) {
+        OptExpression scanExpression = input.getInputs().get(0);
+        Preconditions.checkState(scanExpression.getOp() instanceof LogicalOlapScanOperator);
+        LogicalOlapScanOperator scanOperator = (LogicalOlapScanOperator) scanExpression.getOp();
```

Comment:
please consider group by sub_string(column) case.

Reply:
limit it